lib/geokit-rails/acts_as_mappable.rb

```diff
@@ -137,16 +137,16 @@ def by_distance(options = {})
         bounds  = extract_bounds_from_options(options)
         distance_column_name = distance_sql(origin, units, formula)
         #geo_scope(options).order("#{distance_column_name} asc")
-        order("#{distance_column_name} asc")
+        order("#{distance_column_name} #{options[:reverse] ? 'DESC' : 'ASC'}")
       end
 
       def closest(options = {})
-        by_distance(options).first(1)
+        by_distance(options).limit(1)
       end
       alias nearest closest
 
       def farthest(options = {})
-        by_distance(options).last(1)
+        by_distance({:reverse => true}.merge(options)).limit(1)
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.